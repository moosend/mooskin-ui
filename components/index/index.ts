import '../_utils/globals/global.css';

export {default as Button} from '../Button';
export {default as CheckListItem} from '../CheckListItem';
export {default as CheckboxGroup, Checkbox} from '../Checkbox';
export {default as DatePicker} from '../DatePicker';
export {default as Form, FormGroup} from '../Form';
export {default as FileUpload} from '../File';
export {default as Fieldset} from '../Fieldset';
export {default as Grid, Row, Col} from '../Grid';
export {default as HorizontalRangeBar} from '../HorizontalRangeBar';
export {H1, H2, H3, H4, H5, H6} from '../Headings';
export {default as Input} from '../Input';
export {default as List, ListItem} from '../List';
export {default as LoadingBar} from '../LoadingBar';
export {default as Modal} from '../Modal';
export {default as Pagination} from '../Pagination';
export {default as RadioGroup, Radio} from '../Radio';
export {default as RadioAccordion, RadioAccordionContent} from '../RadioAccordion';
export {default as Steps, Step} from '../Steps';
export {default as SideBar, SidebarItem} from '../SideBar';
export {default as SmallIconButton} from '../SmallIconButton';
export {default as Select, Option} from '../Select';
export {default as Switch} from '../Switch';
export {default as StatsBox, StatsTitle, StatsNumber, StatsResult} from '../StatsBox';
export {default as TextArea} from '../TextArea';
export {default as TabbedContent, Content, Header, Tab} from '../TabbedContent';
export {default as Tags} from '../Tags';
export {default as TopNotification} from '../TopNotification';
export {default as Table, TableHeader} from '../Table';

export {Bar, Doughnut, Pie, Line} from '../Charts';
