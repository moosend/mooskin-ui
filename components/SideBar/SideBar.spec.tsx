import toJson from 'enzyme-to-json';
import * as React from 'react';
import {SideBar, SidebarItem, SmallIconButton} from '../index';

import { mount, shallow } from 'enzyme';

describe('SideBar', () => {

    test('renders SideBar correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                button
            />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders SidebarItem correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SidebarItem
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                active
                href="www.moosend.com"
                label="Item"
                image="imagePath"
                imageOn="imageOnPath"
                onMouseEnter={func}
                onMouseLeave={func}
            />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders SideBar with Items correctly', () => {
        const func = jest.fn();

        const tree = mount(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                button
            >
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
            </SideBar>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders SideBar with Items and secondary SideBar correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                button
            >
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                >
                    <SidebarItem
                        onClick={func}
                        className="myClass"
                        style={{color: 'blue'}}
                        active
                        href="www.moosend.com"
                        label="Item"
                        image="imagePath"
                        onMouseEnter={func}
                        onMouseLeave={func}
                    />
                </SidebarItem>
            </SideBar>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('SideBar prop tests', () => {

        const component = shallow(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                button
            >
                <SidebarItem
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                />
                <SidebarItem
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                />
            </SideBar>
        );

        expect(component.find(SmallIconButton)).toBeTruthy;
        expect(component.find('.sidebar').prop('className')).toContain('myClass');
        expect(component.find('.sidebar').prop('style')).toEqual({color: 'blue'});

        expect(component.find(SidebarItem).length).toEqual(2);
        expect(component.find(SidebarItem).first().prop('className')).toEqual('myClass');
        expect(component.find(SidebarItem).first().prop('style')).toEqual({color: 'blue'});
        expect(component.find(SidebarItem).first().prop('href')).toEqual('www.moosend.com');
        expect(component.find(SidebarItem).first().prop('image')).toEqual('imagePath');
    });

    test('renders different Item images based on active status', () => {

        const component = shallow(
            <SideBar
                button
            >
                <SidebarItem
                    label="Item"
                    image="imagePath"
                    imageOn="imageOnPath"
                />
                <SidebarItem
                    active
                    label="Item"
                    image="imagePath"
                    imageOn="imageOnPath"
                />
            </SideBar>
        );

        expect(component.state('activeItem')).toBe(1);
        expect(component.find('SidebarItem').first().dive().find('img').prop('src')).toEqual('imagePath');
        expect(component.find('SidebarItem').last().dive().find('img').prop('src')).toEqual('imageOnPath');

        component.find('SidebarItem').first().simulate('click');

        expect(component.state('activeItem')).toBe(0);
        expect(component.find('SidebarItem').first().dive().find('img').prop('src')).toEqual('imageOnPath');
        expect(component.find('SidebarItem').last().dive().find('img').prop('src')).toEqual('imagePath');

        component.find('SidebarItem').last().simulate('click');

        expect(component.state('activeItem')).toBe(1);
        expect(component.find('SidebarItem').first().dive().find('img').prop('src')).toEqual('imagePath');
        expect(component.find('SidebarItem').last().dive().find('img').prop('src')).toEqual('imageOnPath');

    });

    test('sidebar hides when hideClick prop is passed and an element is clicked', () => {
        const func = jest.fn();

        const component = mount(
            <SideBar
                className="myClass"
                style={{color: 'blue'}}
                hideClick
                button
            >
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                />
                <SidebarItem
                    onClick={func}
                    className="myClass"
                    style={{color: 'blue'}}
                    active
                    href="www.moosend.com"
                    label="Item"
                    image="imagePath"
                    onMouseEnter={func}
                    onMouseLeave={func}
                >
                    <SidebarItem
                        onClick={func}
                        className="myClass"
                        style={{color: 'blue'}}
                        active
                        href="www.moosend.com"
                        label="Item"
                        image="imagePath"
                        onMouseEnter={func}
                        onMouseLeave={func}
                    />
                </SidebarItem>
            </SideBar>
        );

        expect(component.state('display')).toBe(false);
        expect(component.state('smallDisplay')).toBe(false);
        expect(component.state('subMenuDisplay')).toBe(false);

        expect(component.hasClass('sidebarOff')).toBeTruthy;
        expect(component.hasClass('smallOff')).toBeTruthy;

        expect(component.find('SubMenu').hasClass('subMenuOn')).toBeFalsy;
        expect(component.find('SubMenu').hasClass('sidebarOff')).toBeTruthy;

        component.find('SmallIconButton').simulate('click');

        expect(component.state('display')).toBe(true);
        expect(component.state('smallDisplay')).toBe(true);
        expect(component.state('subMenuDisplay')).toBe(false);

        expect(component.hasClass('sidebarOn')).toBeTruthy;
        expect(component.hasClass('smallOn')).toBeTruthy;

        component.find('SidebarItem').at(1).simulate('mouseenter');

        expect(component.state('display')).toBe(true);
        expect(component.state('smallDisplay')).toBe(true);
        expect(component.state('subMenuDisplay')).toBe(true);

        expect(component.hasClass('sidebarOn')).toBeTruthy;
        expect(component.hasClass('smallOn')).toBeTruthy;

        expect(component.find('SubMenu').hasClass('subMenuOn')).toBeTruthy;
        expect(component.find('SubMenu').hasClass('sidebarOff')).toBeFalsy;

        expect(component.find('SidebarItem').length).toEqual(3);

        component.find('SidebarItem').last().simulate('click');

        expect(component.state('display')).toBe(false);
        expect(component.state('smallDisplay')).toBe(false);
        expect(component.state('subMenuDisplay')).toBe(false);

        expect(component.hasClass('sidebarOff')).toBeTruthy;
        expect(component.hasClass('smallOff')).toBeTruthy;

        expect(component.find('SubMenu').hasClass('subMenuOn')).toBeFalsy;
        expect(component.find('SubMenu').hasClass('sidebarOff')).toBeTruthy;

    });

});
