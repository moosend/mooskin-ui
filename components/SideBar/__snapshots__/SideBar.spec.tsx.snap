// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders SideBar correctly 1`] = `
<div
  className="sidebar-component"
>
  <div>
    <SmallIconButton
      className="button buttonOn undefined"
      icon="view headline"
      onClick={[Function]}
      style={Object {}}
    />
  </div>
  <div
    className="sidebar myClass sidebarOff smallOff"
    style={
      Object {
        "color": "blue",
      }
    }
  />
</div>
`;

exports[`SideBar renders SideBar with Items and secondary SideBar correctly 1`] = `
<div
  className="sidebar-component"
>
  <div>
    <SmallIconButton
      className="button buttonOn undefined"
      icon="view headline"
      onClick={[Function]}
      style={Object {}}
    />
  </div>
  <div
    className="sidebar myClass sidebarOff smallOff"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <SidebarItem
      active={true}
      className="myClass"
      href="www.moosend.com"
      image="imagePath"
      key="0"
      label="Item"
      onClick={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    />
    <SidebarItem
      active={false}
      activeSubMenu={false}
      className="myClass"
      href="www.moosend.com"
      image="imagePath"
      key="1"
      label="Item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    />
  </div>
  <SubMenu
    display={false}
    key=".1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <SidebarItem
      active={false}
      className="myClass"
      href="www.moosend.com"
      label="Item"
      onClick={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    />
  </SubMenu>
</div>
`;

exports[`SideBar renders SideBar with Items correctly 1`] = `
<SideBar
  button={true}
  className="myClass"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="sidebar-component"
  >
    <div>
      <SmallIconButton
        className="button buttonOn undefined"
        icon="view headline"
        onClick={[Function]}
        style={Object {}}
      >
        <button
          className="button-icon-component
                         button
                         normalButton
                         
                         button buttonOn undefined"
          disabled={undefined}
          id={undefined}
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="material-icons icon"
          >
            view_headline
          </i>
        </button>
      </SmallIconButton>
    </div>
    <div
      className="sidebar myClass sidebarOff smallOff"
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <SidebarItem
        active={true}
        className="myClass"
        href="www.moosend.com"
        image="imagePath"
        imageOn={undefined}
        key="0"
        label="Item"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <div
          className="item-component itemContainer myClass"
          onClick={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className=" arrowImg"
          />
          <a
            className="anchor activeItem"
            href="www.moosend.com"
          >
            <img
              className="image"
              src="imagePath"
            />
            <span>
              Item
            </span>
          </a>
        </div>
      </SidebarItem>
      <SidebarItem
        active={false}
        className="myClass"
        href="www.moosend.com"
        image="imagePath"
        imageOn={undefined}
        key="1"
        label="Item"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <div
          className="item-component itemContainer myClass"
          onClick={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className=" arrowImg"
          />
          <a
            className="anchor "
            href="www.moosend.com"
          >
            <img
              className="image"
              src="imagePath"
            />
            <span>
              Item
            </span>
          </a>
        </div>
      </SidebarItem>
    </div>
  </div>
</SideBar>
`;

exports[`SideBar renders SidebarItem correctly 1`] = `
<div
  className="item-component itemContainer myClass"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className=" arrowImg"
  />
  <a
    className="anchor activeItem"
    href="www.moosend.com"
  >
    <img
      className="image"
      src="imageOnPath"
    />
    <span>
      Item
    </span>
  </a>
</div>
`;
